salvo/src/lib/run_benchmark.py

```diff
@@ -0,0 +1,323 @@
+
+    Perform class member initialization and instantiate the underlying
+    object actually performing the test.
+
+    Args:
+      control: The Job Control object dictating the parameters governing the
+      benchmark
+
+    Returns:
+      None
+    """
+    self._control = control
+    self._source_manager = source_manager.SourceManager(self._control)
+
+    self._test = []
+    self._setup_test()
+
+  def _setup_test(self) -> None:
+    """Create the object performing the desired benchmark.
+
+    Instantiate the object performing the actual test and setup the control
+    objects for each test invocation.
+
+    Raises:
+      NotImplementedError: for tests and/or modes that are not yet implemented.
+    """
+
+    current_benchmark_name = "Unspecified Benchmark"
+
+    if self._control.scavenging_benchmark:
+      current_benchmark_name = "Scavenging Benchmark"
+      job_control_list = self.generate_job_control_for_envoy_images()
+
+      for job_control in job_control_list:
+        benchmark = scavenging.Benchmark(job_control, current_benchmark_name)
+        self._test.append(benchmark)
+
+    elif self._control.dockerized_benchmark:
+      current_benchmark_name = "Fully Dockerized Benchmark"
+      job_control_list = self.generate_job_control_for_envoy_images()
+
+      for job_control in job_control_list:
+        benchmark = fulldocker.Benchmark(job_control, current_benchmark_name)
+        self._test.append(benchmark)
+
+    elif self._control.binary_benchmark:
+      current_benchmark_name = "Binary Benchmark"
+      job_control_list = []
+
+    if not self._test:
+      raise NotImplementedError(f"No [{current_benchmark_name}] defined yet")
+
+  def generate_job_control_for_envoy_images(self) \
+      -> List[proto_control.JobControl]:
+    """Determine the required envoy images needed for the benchmark.
+
+    Find the commit hashes or tags for all envoy images, build images if
+    necessary.
+    """
+
+    # Get the images that we are benchmarking. Source Manager will
+    # determine the commit hashes for the images used for benchmarks
+    image_hashes = self._source_manager.get_envoy_hashes_for_benchmark()
+
+    envoy_images = self._pull_or_build_envoy_images_for_benchmark(image_hashes)
+    if not envoy_images:
+      raise Exception("Unable to find or build images for benchmark")
+
+    self._pull_or_build_nighthawk_images_for_benchmark()
+
+    log.debug(f"Using {envoy_images} for benchmark")
+    job_control_list = self.create_job_control_for_images(envoy_images)
+
+    return job_control_list
+
+  def _pull_or_build_nh_benchmark_image(self, \
+                                        images: proto_image.DockerImages) \
+                                        -> None:
+    """Attempt to pull the NightHawk Benchmark Image.  Build the image if
+       unavailable.
+
+       Args:
+        images: the DockerImages appearing in the control object
+    """
+    di = docker_image.DockerImage()
```

Comment:
Can we find a longer and more descriptive name instead of just `di`? Maybe `docker_image` ?

https://github.com/google/styleguide/blob/gh-pages/pyguide.md#22-imports

Applies to multiple locations.

Reply:
Updated